Move to the github.com/qpliu/qrencode-go/qrencode package

zaphar opened the pull request:
This package is the most used and starred on gosearch which
is as good a proxy as I could think of for likelihood of support
going forward.

I test generated a png and verified that it was importable into
google authenticator as well as the popular DUO authenticator.

It's a little more work in our code to generate the barcode but not that much really.

Fixes #1

zaphar committed:
Move to the github.com/qpliu/qrencode-go/qrencode package

This package is the most used and starred on gosearch which
is as good a proxy as I could think of for likelihood of support
going forward.

I test generated a png and verified that it was importable into
google authenticator as well as the popular DUO authenticator.

balasanjay commented on totp.go:
```diff
@@ -38,13 +42,15 @@ func BarcodeImage(label string, secretkey []byte, opt *Options) ([]byte, error)
 	}
 
-	return c.PNG(), nil
+	buf := bytes.NewBuffer([]byte{})
```
The zero-value of bytes.Buffer is usable. i.e. this can simply be "var buf bytes.Buffer" (and add an ampersand to the next line).

balasanjay commented on totp.go:
```diff
@@ -38,13 +42,15 @@ func BarcodeImage(label string, secretkey []byte, opt *Options) ([]byte, error)
 
-	return c.PNG(), nil
+	buf := bytes.NewBuffer([]byte{})
+	err = png.Encode(buf, c.Image(8))
```
I don't know if this is possible, but if png.Encode writes a partial result to the buffer, then returns an error, then we'll return a non-nil buffer and a non-nil error. To preserve the previous behavior (or atleast to make it obvious that this can't happen), can you add another error check here?

balasanjay commented:
Thanks! Just a couple of nits, and this looks good to me.

Also, I think if you put "Fixes #1" in the PR description, it'll link and auto-close that issue.

zaphar committed:
Check the error from the write into the buffer.

Technically I don't think it's even possible for it to error
there. But to avoid potential gotcha's it's a good idea
to check the error and preserve the former interface.

zaphar committed:
Cleanups and feedback from balasanjay

Use the zero value of bytes.Buffer.

Check the error from the write into the buffer.

Technically I don't think it's even possible for it to error
there. But to avoid potential gotcha's it's a good idea
to check the error and preserve the former interface.

zaphar committed:
accidentally forked my own branch merge it back in.

Because I suck at git.

zaphar commented:
Feedback addressed. Please take another look.

balasanjay commented:
LGTM. Thanks again!
